app/src/main/java/com/github/orkest/Model/FireStoreDatabaseAPI.kt

```diff
@@ -273,4 +272,53 @@ class FireStoreDatabaseAPI {
 
         return future
     }
+
+    /**
+     * This method stores a shared song to the database of the receiver
+     *
+     * @param song the song we want to add in the database
+     * @param receiver the receiver of the shared song
+     *
+     * @return a completable future that indicates that the song was correctly added
+     */
+    fun storeSharedSongToDataBase(song: Song,sender: String, receiver: String): CompletableFuture<Boolean> {
+        val future = CompletableFuture<Boolean>()
+        val sharedSongDocument1 =
+            db.collection("shared songs")
+                .document(receiver)
+                .collection(sender)
+                .document()
+
+        sharedSongDocument1.set(song).addOnSuccessListener {
+            future.complete(true)
+        }.addOnFailureListener {
+                future.completeExceptionally(it)
+            }
+        return future
+    }
+
+    /**
+     * Fetch songs shared with current user from a sender
+     * @param senderUsername the sender of the shared song
+     *
+     * @return a completable future of a list of post that were recently published
+     */
+    fun fetchSharedSongsFromDataBase(senderUsername: String, receiverUsername: String): CompletableFuture<List<Song>> {
+        val future = CompletableFuture<List<Song>>()
+
+        val sharedSongs = db.collection("shared songs").document(receiverUsername).collection(senderUsername)
```

Comment:
This val could be refactored into a private function like for the userDocumentRef and PostDocumentRef since it is used 2 times and would improve maintainability for future codes in case the path to the sharedSongs would change